src/quantity.rs

```diff
@@ -210,6 +210,54 @@ macro_rules! quantity {
             impl super::Conversion<V> for super::$unit {})+
         }
 
+        #[cfg(feature = "std")]
+        mod parse {
+            use $crate::lib::str::FromStr;
+
+            /// Represents an error encountered while parsing a string into a `Quantity`.
+            #[derive(Clone, Copy, Debug)]
+            pub enum ParseError {
+                /// No separators (spaces) were encountered.
+                NoSeparator,
+                /// An error occured while parsing the value (first) portion of the string.
+                ///
+                /// Due to exhaustiveness and type system limitations, this variant does not encode
+                /// the underlying parse error.
+                ValueParseError,
+                /// The abbreviation used wasn't found for this quantity.
+                UnknownAbbreviation,
+            }
+
+            impl<U, V> FromStr for super::$quantity<U, V>
+            where
+                U: super::super::Units<V> + ?Sized,
+                V: $crate::num::Num + $crate::Conversion<V> + FromStr,
+                $(
+                    super::$unit: $crate::Conversion<V, T = V::T>,
+                )+
+            {
+                type Err = ParseError;
+                #[allow(unreachable_patterns)]
+                fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
Will need to investigate this in more detail. May be something where this implementation can be accepted and then optimized later (finite state table vs. `match`?).

 * `contains` and `split` cause part of the `&str` to be iterated over multiple times. What about finding he index of the first space and then extracting the two sub-slices.
 * What if an abbreviation has spaces in it? Won't this only attempt to parse the first portion as the abbreviation?

Reply:
I think that using [`splitn`](https://doc.rust-lang.org/std/primitive.str.html#method.splitn) will resolve these concerns.